src/olympia/hero/models.py

```diff
@@ -97,16 +99,33 @@ class PrimaryHeroImage(ModelBase):
     def __str__(self):
         return f'{self.custom_image}'
 
+    @property
+    def preview_url(self):
+        (path, fn) = os.path.split(self.custom_image.path)
+        fn = fn.decode('utf-8')
+        return f'{HERO_PREVIEW_URL}{fn}'
+
     def preview_image(self):
         if self.custom_image:
             return mark_safe('<img class="prmhero-preview" src="{}" />'.format(
-                self.custom_image.url)
+                self.preview_url)
             )
         else:
             return None
     preview_image.short_description = "Image"
     preview_image.allow_tags = True
 
+    def save(self, *args, **kwargs):
+        super().save(*args, **kwargs)
+        (path, fn) = os.path.split(self.custom_image.path)
+        dest_thumb = path + b'/thumbs/' + fn
+
+        size_full = (960, 640)
+        size_thumb = (150, 120)
+
+        resize_image(self.custom_image, self.custom_image.path, size_full)
```

Comment:
the first arg here, `self.custom_image` is just the image path, isn't it?  (i.e. the same as the second arg).  Tbh I don't know why we enforce that the source and destination be different in `resize_image` - you might be introducing a bug you have to fix later 😉  I'd recommend you write to a temporary file and then move it back but I'm not going to require it.

Reply:
The two arguments are different. The first argument, `self.custom_image`, has an `ImageFieldFile` value, while the second argument (self.custom_image.path) has a bytes object value (see example below). Please let me know if I should still write to a temporary file.
<img width="877" alt="Screen Shot 2020-06-30 at 1 16 35 PM" src="https://user-images.githubusercontent.com/28129806/86156437-f5078d00-bad3-11ea-9771-ba8db8100960.png">

Reply:
Definitely don't want bugs 😬 I'm not 100% sure about my revision though, can you please check if I incorporated the temporary file correctly?